apps/core/models.py

```diff
@@ -10,8 +11,7 @@ class DatasetStatus(models.TextChoices):
         UPLOADED = "uploaded", _("Uploaded")
         EXTRACTED = "extracted", _("Extracted")
 
-    file_url = models.TextField()
-    file_size_bytes = models.PositiveIntegerField()
+    file = models.ForeignKey(File, on_delete=models.SET_NULL, null=True)
```

Comment:
Are we preserving the dataset once the file gets deleted?

Reply:
Hmm.. that's a fuzzy borderline there. I don't actually remember why I kept it null or even if I did that. But I think dataset can have a separate existence from file. Plus, dataset contains a lot of other information like tables and columns and snapshots(in near future) so we can just keep it nullable and delete the dataset manually later if needed.